src/application-navigation/terra-application-navigation/common/_PopupMenu.jsx

```diff
@@ -0,0 +1,217 @@
+import React, {
+  useRef,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import ActionFooter from 'terra-action-footer';
+import ContentContainer from 'terra-content-container';
+import Button from 'terra-button';
+import ActionHeader from 'terra-action-header';
+import {
+  KEY_SPACE,
+  KEY_RETURN,
+  KEY_UP,
+  KEY_DOWN,
+} from 'keycode-js';
+
+import PopupMenuListItem from './_PopupMenuListItem';
+import { userConfigPropType } from '../utils/propTypes';
+import PopupMenuUser from './_PopupMenuUser';
+
+import styles from './PopupMenu.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Title to be placed within the popup header.
+   */
+  title: PropTypes.string,
+  /**
+   * The text to be assigned to the button in the footer.
+   */
+  footerText: PropTypes.string,
+  /**
+   * Callback for when the footer item is selected.
+   */
+  onSelectFooterItem: PropTypes.func,
+  /**
+   * The custom content to be placed at the top of the content area.
+   */
+  customContent: PropTypes.node,
+  /**
+   * A configuration object with information pertaining to the application's user.
+   */
+  userConfig: userConfigPropType,
+  /**
+   * An array of selectable items to be displayed within the popup.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     * Data attributes to identify the rendered menuItem.
+     */
+    dataAttrs: PropTypes.object,
+    /**
+     * Icon to be rendered
+     */
+    icon: PropTypes.element,
+    /**
+     * Key used as react key, and returned in the onSelect
+     */
+    key: PropTypes.string.isRequired,
+    /**
+     * Object to be returned in the onSelect
+     */
+    metaData: PropTypes.object,
+    /**
+     * Text display and/or aria-label
+     */
+    text: PropTypes.string.isRequired,
+    /**
+     * Whether or not this item is the active item.
+     */
+    isActive: PropTypes.bool,
+    /**
+     * The number value representing the notification count.
+     */
+    notificationCount: PropTypes.number,
+  })),
+  /**
+   * Function callback for selection of the menu item. Returns (key, metaData).
+   */
+  onSelectMenuItem: PropTypes.func,
+  /**
+   * Whether or not content is constrained vertically.
+   */
+  isHeightBounded: PropTypes.bool,
+  /**
+   * Whether or not selected states should display on the menu item.
+   */
+  showSelections: PropTypes.bool,
+  /**
+   * Role of the parent ul that child items should match.
+   */
+  role: PropTypes.oneOf(['list', 'menu', 'listbox']),
+};
+
+const defaultProps = {
+  menuItems: [],
+  isHeightBounded: false,
+  showSelections: false,
+  role: 'listbox',
+};
+
+const PopupMenu = ({
```

Comment:
I know this is a migration from terra-framework, but I'm wondering if we should add a high level explanation on more complicated packages that have multiple sub-components in it's implementation

Reply:
I do believe this would be helpful. I can take a stab at this when I make my changes to the Navigation component to support our new architecture.